src/CommunicationExecutor.c

```diff
@@ -268,6 +325,12 @@ void Ros_Communication_StartExecutors(SEM_ID semCommunicationExecutorStatus)
     rc = rclc_executor_add_timer(&executor_motion_control, &timerPublishActionFeedback);
     motoRosAssert_withMsg(rc == RCL_RET_OK, SUBCODE_FAIL_TIMER_ADD_ACTION_FB, "Failed adding timer (%d)", (int)rc);
 
+    //NOTE: add userlan monitor timer to the io executor, to prevent timerPingAgent
+    //from blocking it in case agent connection is lost (ie: ping needs to time out)
+    rc = rclc_executor_add_timer(&executor_io_control, &timerMonitorUserLanState);
```

Comment:
moved the timer to the io executor (from the motion executor). Both to not block motion callbacks while it runs and also to avoid the ping Agent callback from blocking execution of the lan state monitor callback.